python-backend/graph_utils.py

```diff
@@ -22,103 +21,18 @@ class Point2D(TypedDict):
     y: str
 
 
-async def chart_coordinates(values: dict[int, mpmath.mpf],
-                            p_values: dict[int, mpmath.mpf],
-                            q_values: dict[int, mpmath.mpf],
-                            limit: mpmath.mpf,
-                            iterations: int, websocket: WebSocket) -> tuple[
-    list[Point2D], list[Point2D], list[Point2D], list[Point2D]]:
+async def chart_coordinates(pcf: pcf_module.PCF, limit: mpmath.mpf,
+                            iterations: int, websocket: WebSocket) -> None:
     """
     graph coords for the error of an expression: |expression - L|
-    :param values: computed values for continued fraction, p_n/q_n
-    :param p_values: computed values for continued fraction numerators, p_n
-    :param q_values: computed values for continued fraction denominators, q_n
+    :param pcf: PCF instance
     :param limit: the computed approximate limit of the expression
     :param iterations: the number of values expected, n max
+    :param websocket: the websocket instance to return incremental data
     :return: tuple of arrays of [x,y] pairs for graphing purposes
     """
-    error_x_y_pairs = []
```

Comment:
all the calculations happen in LIReC and ResearchTools so none of this is needed anymore